### PR TITLE
[telemetry] Fix checks_entered gauge

### PR DESCRIPTION
**Pull request by olivielpeau**
### What does this PR do?

Decrements the gauge when a check is removed from the scheduler. Fixes
ever increasing gauge when AD checks are scheduled/unscheduled.

### Motivation

Fix telemetry metric.

### Additional Notes

* Went for a crude map to track check names by ID, for the purpose of telemetry only
* internal feature, so no changelog

### Describe your test plan

I tested in AD scenario, with telemetry enabled, including details on checks (`telemetry.enabled` and `telemetry.checks`, see `config.go` for details).

1. check `curl http://localhost:5000/telemetry | grep scheduler`
2. start a container that will start an AD-scheduled check
3. check `curl http://localhost:5000/telemetry | grep scheduler` again, the new check should show up, with value of `1`
4. stop the container, the AD-scheduled check should stop, check that the related value is `0` in output of `curl http://localhost:5000/telemetry | grep scheduler` (this is what the PR fixes, with an older agent the value would remain `1`)

### CODE REVIEW

File: pkg/collector/scheduler/scheduler.go [hunk 1]
```diff
@@ -126,6 +130,9 @@ func (s *Scheduler) Cancel(id check.ID) error {
 	delete(s.checkToQueue, id)
 
 	schedulerChecksEntered.Add(-1)
+	if checkName, ok := s.tlmTrackedChecks[id]; ok {
```
Comment (truthbk) [inline pkg/collector/scheduler/scheduler.go:133]: Should the check also be removed from `tlmTrackedChecks`?

**Review by truthbk — COMMENTED** [review]
Looks good, just a question.

File: pkg/collector/scheduler/scheduler.go (hunk 1 above)
Comment (olivielpeau) [reply pkg/collector/scheduler/scheduler.go:133]: you're right, I'll address

Comment (olivielpeau) [reply pkg/collector/scheduler/scheduler.go:133]: addressed, ready for another review @truthbk

**Review by truthbk — APPROVED** [review]
📦

**Pull request merged**